### DISCUSSION TITLE
Browser is slow when load multiple large file

### DISCUSSION THREAD

**Discussion by sonphambk** [visgl/deck.gl#6257] [Q&A]
HI.
I tried to use deck.gl to load PCD file. When I uploaded 1 file with size about 200MB, it is smooth. But when I tried to load multi PCD file (about 300MB). PCD is very slow when I zoom in and zoom out, drag left or right.
Can you help me?
Thanks.

**Comment by Pessimistress** [comment]
There is a limit to how many points your browser/GPU can handle at once. To render very large point clouds or meshes you need to use the `Tile3DLayer`, which only loads/renders geometries that are required by the current viewport. This requires you to convert your data to one of the 3D tiling standards (3D Tiles or I3S).

See examples here:

https://deck.gl/examples/tile-3d-layer/

https://deck.gl/examples/arcgis/

**Comment by ibgreen** [comment]
Agreed, without more information it seems most likely that you are simply maxing out your GPU. Check the total number of points you are rendering (sum over all the parsed meshes). With good GPU you can may be able to render up to a few million points relatively smoothly, but tens of million of points will typically be too much. Maybe try to load fewer PCD files until you find the sweet spot.
